### PR TITLE
COVsearch update

### PR DESCRIPTION
**Pull request by JohanWBorg**
Implemented covariate as a part of ModelFeatures and changes to covsearch for handling of already present covariates
- Covariate attribute is now a part of ModelFeautures, allowing addition, subtraction and least number of transformation between different models/search spaces
- Possibility of extracting covariates from a model of choosing, with effect (based on our effect templates) and operator using "get_covariates()". Also implemented this feature within "get_model_features()"
- Changes to covsearch to handle already present covariates. If present covariates is not part of search space, they are removed before running covsearch. If they are a part of the model and the search space, they are left as is and are removed from the search space.

### CODE REVIEW

**Comment by rikardn** [comment]
@JohanWBorg Could you have a look at the conflicts?

**Comment by JohanWBorg** [comment]
@rikardn Conflicts should be resolved now

File: src/pharmpy/modeling/covariate_effect.py [hunk 1]
```diff
@@ -14,13 +15,116 @@
...
+    remove_covariate_effect_from_statements,
+    simplify_model,
+)
+from .parameters import get_thetas
 
 EffectType = Union[Literal['lin', 'cat', 'piece_lin', 'exp', 'pow'], str]
 OperationType = Literal['*', '+']
 
 
+def get_covariates(model: Model):
```
Comment (rikardn) [inline src/pharmpy/modeling/covariate_effect.py:32]: Need return type hint for the pharmr code generation

File: src/pharmpy/tools/mfl/parse.py [hunk 2]
```diff
@@ -703,13 +707,31 @@ def get_model_features(model: Model, supress_warnings: bool = False) -> str:
         transits = f'TRANSITS({transits}{","+depot})'
     if peripherals != 0:
         peripherals = f'PERIPHERALS({peripherals})'
+    if len(covariates) != 0:
+        # if any([v[0][0] == 'custom' for _, v in covariates.items()]):
```
Comment (rikardn) [inline src/pharmpy/tools/mfl/parse.py:711]: Remove commented code

**Review by rikardn — CHANGES_REQUESTED** [review]

**Pull request merged**